File: Juan/P4/Java/Napakalaki/src/NapakalakiGame/prueba.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package NapakalakiGame;

/**
 *
 * @author Raul
 */
public class prueba extends Absprueba implements Interfaceprueba {
    
    prueba(){
        super.valor=3;
    }
    
    @Override
    public void diHola(){
        super.diHola();
        System.out.print(" tu culo no mola" + numero);
    }
    
    @Override
    public void diAdios(){
        super.diAdios();
        System.out.print(" tu culo no mola");
    }
}

File: Juan/P4/Ruby/Napakalaki/lib/Prueba.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'Absprueba.rb'

module NapakalakiGame
    class Prueba < Absprueba #Herencia
        include Interfaceprueba #Interface

        def initialize(caca)
            @algo=caca
        end

        public_class_method :new #necesario para poder accder a la clase abstracta
        
        def diHola
            super
            puts ' caracola'
        end
    end
end
